v2/spiffetls/tlsconfig/config.go

```diff
@@ -26,63 +26,92 @@ func HookTLSClientConfig(config *tls.Config, bundle x509bundle.Source, authorize
 	config.VerifyPeerCertificate = WrapVerifyPeerCertificate(config.VerifyPeerCertificate, bundle, authorizer)
 }
 
+// A Option changes the defaults used to by mTLS ClientConfig functions.
+type Option interface {
```

Comment:
I think we're missing passing the options into a few exported functions in this package (i.e. HookTLSClientConfig, etc). Even though they won't exercise the current trace callbacks, they will soon and we don't want to have to revise the exported function signatures. Can we update those functions to also take the variadic slice of options?